### PR TITLE
Adding promptflow-evals

### PR DESCRIPTION
**Pull request by singankit**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by ninghu — APPROVED** [review]

**Review by nagkumar91 — APPROVED** [review]

File: src/promptflow-evals/promptflow/evals/evaluators/similarity/flow/validate_inputs.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from promptflow import tool
```
Comment (wangchao1230) [inline src/promptflow-evals/promptflow/evals/evaluators/similarity/flow/validate_inputs.py:1]: please use promptflow.core import tool

File: src/promptflow-evals/promptflow/evals/evaluators/similarity/flow/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+promptflow
```
Comment (wangchao1230) [inline src/promptflow-evals/promptflow/evals/evaluators/similarity/flow/requirements.txt:1]: promptflow -> promptflow-core?

File: src/promptflow-evals/promptflow/evals/evaluate/_flow_run_wrapper.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+import time
+from promptflow import PFClient
```
Comment (wangchao1230) [inline src/promptflow-evals/promptflow/evals/evaluate/_flow_run_wrapper.py:5]: from promptflow.client import PFClient

File: src/promptflow-evals/promptflow/evals/evaluators/chat/__init__.py [hunk 4]
```diff
@@ -0,0 +1,220 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+__path__ = __import__("pkgutil").extend_path(__path__, __name__)  # type: ignore
+
+from promptflow.entities import AzureOpenAIConnection
```
Comment (wangchao1230) [inline src/promptflow-evals/promptflow/evals/evaluators/chat/__init__.py:7]: from promptflow.connections import AzureOpenAIConnection

File: src/promptflow-evals/promptflow/evals/evaluators/coherence/__init__.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+__path__ = __import__("pkgutil").extend_path(__path__, __name__)  # type: ignore
+
+from promptflow import load_flow
```
Comment (wangchao1230) [inline src/promptflow-evals/promptflow/evals/evaluators/coherence/__init__.py:7]: from promptflow.client import load_flow

File: src/promptflow-evals/promptflow/evals/evaluators/coherence/flow/validate_inputs.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+from promptflow import tool
```
Comment (wangchao1230) [inline src/promptflow-evals/promptflow/evals/evaluators/coherence/flow/validate_inputs.py:1]: please do promptflow.core import tool 
same other places

File: src/promptflow-evals/promptflow/evals/evaluators/coherence/flow/requirements.txt [hunk 7]
```diff
@@ -0,0 +1,2 @@
+promptflow
```
Comment (wangchao1230) [inline src/promptflow-evals/promptflow/evals/evaluators/coherence/flow/requirements.txt:1]: please change dag flow requirements:
promptflow -> promptflow-core

same other places

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow-evals/requirements.txt [hunk 8]
```diff
@@ -0,0 +1,4 @@
+azure-ai-ml>=1.14.0
+promptflow
```
Comment (wangchao1230) [inline src/promptflow-evals/requirements.txt:2]: depends on promptflow-core or promptflow-devkit?

File: src/promptflow-evals/promptflow/evals/evaluators/similarity/flow/validate_inputs.py (hunk 1 above)
Comment (singankit) [reply src/promptflow-evals/promptflow/evals/evaluators/similarity/flow/validate_inputs.py:1]: Working on a follow up PR to resolve this comment.

**Pull request merged**